src/plugins/keras_support/categorical_crossentropy_operation.cpp

```diff
@@ -71,11 +71,39 @@ namespace phylanx { namespace execution_tree { namespace primitives
 
     primitive_argument_type cat_cross_operation::cat_cross0d() const
     {
-        return primitive_argument_type{static_cast<double>(1.)};
+        /*
+        if(from_logits)
+        {
+            // softmax0d is 1
+            output.scalar() = blaze::softmax(output.scalar());
+        }
+        else
+        {
+            // divide by itself is 1
+            output.scalar() /= blaze::sum(output.scalar());
+        }
+        // regardless of from_logits, output = 1
+
+        // Construct low and high clip regions
+        primitive_argument_type lo_{small}, hi_{1-small};
+
+        auto lo = extract_value_scalar<double>(lo_, name_, codename_ );
+        auto hi = extract_value_scalar<double>(hi_, name_, codename_ );
+
+        output.scalar() = blaze::max(blaze::min(
+            output.scalar(), hi.scalar()), lo.scalar());
+        // output is still 1 after clipping
+
+        // log of 1 is zero, so res is zero
+        auto res = target.scalar() * (-blaze::log(output.scalar()));
+        return primitive_argument_type{ blaze::sum(res) };
+        */
+        return primitive_argument_type{static_cast<double>(0.)};
```

Comment:
I think we better to consider this case invalid since other other backends don't consider it.
```py
x = K.constant(1)
y = K.constant(0)
a = K.categorical_crossentropy(x,y)
K.get_value(a)
```
```pytb
ValueError: Invalid reduction dimension -1 for input with 0 dimensions. for 'Sum_2' (op: 'Sum') with input shapes: [], [] and with computed input tensors: input[1] = <-1>.
```
I think mapping 0d loss to zero is not correct because having a 0d array, we can have two classes; the loss should be different when the prediction (output) is close to the target and when it is not.